### PR TITLE
DRYied scrapers by removing who_scraper since it's refactored by @luizvarela

### PR DESCRIPTION
**Pull request by AfolabiOlaoluwa**
Already fixed in @luizvarela PR

### CODE REVIEW

**Review by luizvarela — APPROVED** [review]

**Comment by luizvarela** [comment]
But you need to remove the specs

**Comment by AfolabiOlaoluwa** [comment]
@luizvarela yes I have removed its spec as well. Bear with me am on mobile.

File: app/lib/scrapers/who_scraper.rb [hunk 1]
```diff
@@ -1,64 +0,0 @@
...
-          next
-        end
-      end
-    end
-    disease_data
-  end
-
-  def facts(disease_page)
-    parsed_lists = disease_page.at('h3:contains("Key facts")').try(:next_element)
-    if parsed_lists && (lists = parsed_lists.search("li")).present?
```
Comment (x6iae) [inline app/lib/scrapers/who_scraper.rb:59]: Has this been re-implemented in the re-factor work by Luiz?

Comment (luizvarela) [reply app/lib/scrapers/who_scraper.rb:59]: Yes, you can check here https://github.com/devcenter-square/disease-info/blob/develop/app/lib/scrapers/who/disease_parser.rb#L50

**Comment by x6iae** [comment]
One more thing:

I think we'll need to link this up to the new class. https://github.com/devcenter-square/disease-info/blob/develop/lib/tasks/mine_data.rake#L4

cc: @AfolabiOlaoluwa @luizvarela

**Comment by luizvarela** [comment]
@SundayAdefila You're right, we need to change it to use the new class.
would be good a test for this. :+1:

**Comment by luizvarela** [comment]
Inside of data_miner.rb, change for:

```

def get_who_data
  Scrapers::Who::Parser.perform
end

```

https://github.com/devcenter-square/disease-info/blob/develop/app/lib/data_miner.rb

**Comment by AfolabiOlaoluwa** [comment]
@luizvarela @SundayAdefila, I suppose this passed in the necessary class for https://github.com/devcenter-square/disease-info/blob/develop/lib/tasks/mine_data.rake#L4

**Comment by AfolabiOlaoluwa** [comment]
@luizvarela, you might have to look at my exceptions. `Scrapers::Who::DiseaseParser does not implement: data`

File: app/lib/data_miner.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ module DataMiner
   extend self
 
   def get_who_data
-    Scrapers::WhoScraper.get_data
+    Scrapers::Who::DiseaseParser.get_data
   end
```
Comment (luizvarela) [inline app/lib/data_miner.rb:8]: You should call  `Scrapers::Who::Parser.perform` instead.

File: spec/lib/data_miner_spec.rb [hunk 3]
```diff
@@ -2,9 +2,9 @@
 
 describe DataMiner do
   describe "#get_who_data" do
-    it "should call the #get_data method of Scrapers::WhoScraper" do
-      expect(Scrapers::WhoScraper).to receive(:get_data)
+    it "should call the #data method of Scrapers::Who::DiseaseParser" do
+      allow(Scrapers::Who::DiseaseParser).to receive(:data)
       DataMiner.get_who_data
```
Comment (luizvarela) [inline spec/lib/data_miner_spec.rb:7]: And you need to change to `allow(Scrapers::Who::Parser).to receive(:perform)`.

File: app/lib/data_miner.rb (hunk 2 above)
Comment (luizvarela) [reply app/lib/data_miner.rb:8]: The `Scrapers::Who::Parser` is responsible for saving all diseases in the database

**Review by luizvarela — APPROVED** [review]

**Comment by x6iae** [comment]
@AfolabiOlaoluwa can you rebase develop into this branch so as to get travis passing? and also confirm that after switching, and running the rake task, the database got populated with diseases?

**Comment by AfolabiOlaoluwa** [comment]
@luizvarela @SundayAdefila I suppose you two can review and Approve this now. Just go through in case anything is oversight for me. We are cool to merge as I see. Nevertheless, thanks to both of you.

File: app/lib/data_miner.rb (hunk 2 above)
Comment (AfolabiOlaoluwa) [reply app/lib/data_miner.rb:8]: @luizvarela You are right. See the PR now

**Review by AfolabiOlaoluwa — COMMENTED** [review]
Its a go

**Review by luizvarela — APPROVED** [review]

**Comment by x6iae** [comment]
Cool cool and cool... weldone y'all and me hehe... 😄 
Merging this.

**Pull request merged**